rollup/internal/controller/relayer/l2_relayer.go

```diff
@@ -729,6 +714,140 @@ func (r *Layer2Relayer) handleL2RollupRelayerConfirmLoop(ctx context.Context) {
 	}
 }
 
+func (r *Layer2Relayer) constructCommitBatchPayloadCodecV0(dbBatch *orm.Batch, dbParentBatch *orm.Batch, dbChunks []*orm.Chunk, chunks []*encoding.Chunk) ([]byte, error) {
+	daBatch, err := codecv0.NewDABatchFromBytes(dbBatch.BatchHeader)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create DA batch from bytes: %w", err)
+	}
+
+	encodedChunks := make([][]byte, len(dbChunks))
+	for i, c := range dbChunks {
+		daChunk, createErr := codecv0.NewDAChunk(chunks[i], c.TotalL1MessagesPoppedBefore)
+		if createErr != nil {
+			return nil, fmt.Errorf("failed to create DA chunk: %w", createErr)
+		}
+		daChunkBytes, encodeErr := daChunk.Encode()
+		if encodeErr != nil {
+			return nil, fmt.Errorf("failed to encode DA chunk: %w", encodeErr)
+		}
+		encodedChunks[i] = daChunkBytes
+	}
+
+	calldata, packErr := r.l1RollupABI.Pack("commitBatch", daBatch.Version, dbParentBatch.BatchHeader, encodedChunks, daBatch.SkippedL1MessageBitmap)
+	if packErr != nil {
+		return nil, fmt.Errorf("failed to pack commitBatch: %w", packErr)
+	}
+	return calldata, nil
+}
+
+func (r *Layer2Relayer) constructCommitBatchPayloadCodecV1(dbBatch *orm.Batch, dbParentBatch *orm.Batch, dbChunks []*orm.Chunk, chunks []*encoding.Chunk) ([]byte, *kzg4844.Blob, error) {
+	batch := &encoding.Batch{
+		Index:                      dbBatch.Index,
+		TotalL1MessagePoppedBefore: dbChunks[len(dbChunks)-1].TotalL1MessagesPoppedBefore + dbChunks[len(dbChunks)-1].TotalL1MessagesPoppedInChunk,
```

Comment:
Isn't this popped after?

Reply:
Fixed in [0d72756](https://github.com/scroll-tech/scroll/pull/1201/commits/0d7275644803acdcfd3a006cd73c44d3e51edf42).